### PR TITLE
[Bugfix:Submission] Fix Notebook Display bugs

### LINKED ISSUE #5633: Notebook display and boolean gradeable options
**Describe the bug**
When "Should submissions be accepted after the due date?" is false, we don't display the notebook.
Students cannot see their submission (and cannot study the grading when grades are released)
note: instructors can still see this display -- why?

Also, if a student has an excused absence extension they cannot view or submit [buggy] -- they should be able to do so.

**Expected behavior**
We need to debug all combinations of boolean options for the display & submission.
And add test cases.

These settings should control what's displayed:
View the gradeable on the course home page and download submitted files?
Make new submissions and access all prior versions?

"Should submissions be accepted after the due date?" should only control making a new submission only.

### PR DESCRIPTION
**Pull request by loadedskiffer**
### What is the current behavior?
closes #5633 

### What is the new behavior?
1. Students can see their notebook after submissions are closed even if accept late submissions=no
2. Students with extensions will be able to see and submit their notebook if submissions are closed.
3. if a student has an extension but accept late submissions = no, they still won't be able to submit
4. A nicer (not red) message is displayed if students are trying to check their grade on a notebook and accept late submissions=no

### NOTE:
you may run into multiple twig exceptions when testing, I am going to go through and fix them but if you test this before that happens, just delete the lines throwing exceptions because they are irrelevant to this pr.

To test:
1. Close a notebook gradeable in the dev course
2. toggle accept late submissions and see different behavior 
3. give students extensions and see behavior

### CODE REVIEW

File: site/app/templates/submission/homework/SubmitBox.twig [hunk 1]
```diff
@@ -312,7 +311,10 @@
                             Drag your file(s) here or click to open file browser
                         {% endif %}
                     </h2>
-                    <input type="file" name="files" id="input-file{{ loop.index }}" class="hide" onchange="addFilesFromInput({{ loop.index }})" multiple aria-label="Select Files to upload" />
+                    <input type="file" name="files" id="input-file{{ loop.index }}" class="hide" onchange="addFilesFromInput({{ loop.index }})" multiple aria-label="Select Files to upload"
```
Comment (williamjallen) [inline site/app/templates/submission/homework/SubmitBox.twig:314]: you could use a ternary operator here if you want to make it fit on one line

**Review by williamjallen — APPROVED** [review]

**Review by bmcutler — CHANGES_REQUESTED** [review]
I think it looks good, but I'm not ready to approve this because I'm getting stuck testing the manual grading of notebook gradeables.  I get this twig error:

Screen Shot 2021-07-21 at 1 52 31 AM

(seems like it is also broken on master)

**Comment by bmcutler** [comment]
another twig error:

Screen Shot 2021-07-27 at 10 24 10 AM

I got this clicking on ta grading for Notebook basic for the instructor user, which I had submtited to once.

It looks like it might be related to @DefinitiveAbove recent work?

**Review by williamjallen — CHANGES_REQUESTED** [review]
Code looks good.  I was able to produce the following Twig error by following the below sequence:
1. Closing the notebook_time_limit gradeable
2. Giving the user "student" a 1-day extension
3. Viewing the gradeable from the student account

Untitled

**Review by williamjallen — APPROVED** [review]

**Pull request merged**